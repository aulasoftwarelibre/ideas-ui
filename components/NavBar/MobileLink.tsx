import Link from 'next/link';
import { NextRouter, withRouter } from 'next/router';
import React from 'react';

interface Props {
  label: string;
  path: string;
  router: NextRouter;
}

const MobileLink: React.FunctionComponent<Props> = ({
  label,
  path,
  router,
}) => {
  const isActive: boolean = router.pathname == path;

  if (isActive) {
    return (
      <span className="mt-1 block px-3 py-2 rounded-md text-base font-medium text-white bg-ideaOrange focus:outline-none focus:text-white focus:bg-ideaOrange transition duration-150 ease-in-out">
        {label}
      </span>
    );
  }

  return (
    <Link href={path}>
      <a className="mt-1 block px-3 py-2 rounded-md text-base font-medium text-ideaOrange hover:text-white hover:bg-ideaOrange focus:outline-none focus:text-white focus:bg-ideaOrange transition duration-150 ease-in-out">
        {label}
      </a>
    </Link>
  );
};

export default withRouter(MobileLink);
