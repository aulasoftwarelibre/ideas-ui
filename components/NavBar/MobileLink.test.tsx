import * as React from 'react';

import { mountMockRouter, renderMockRouter } from '../../utils/test-utils';
import MobileLink from './MobileLink';

describe("MobileLink", () => {
  it("should be rendered correctly", () => {
    const { asFragment } = renderMockRouter(
      <MobileLink path="/no-active" label="Profile" />,
      {
        router: { pathname: "/active" },
      }
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should be rendered correctly when current is enabled", () => {
    const { asFragment } = renderMockRouter(
      <MobileLink path="/active" label="Profile" />,
      {
        router: { pathname: "/active" },
      }
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should be a link", () => {
    const wrapper = mountMockRouter(
      <MobileLink path="/active" label="Profile" />,
      {
        router: { pathname: "/" },
      }
    );

    expect(wrapper.find("a")).toHaveLength(1);
  });

  it("should be a span when active", () => {
    const wrapper = mountMockRouter(
      <MobileLink path="/active" label="Profile" />,
      {
        router: { pathname: "/active" },
      }
    );

    expect(wrapper.find("span")).toHaveLength(1);
  });
});
