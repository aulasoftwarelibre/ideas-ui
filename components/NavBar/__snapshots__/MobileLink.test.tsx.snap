// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileLink should be rendered correctly 1`] = `
<DocumentFragment>
  <a
    class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-ideaOrange hover:text-white hover:bg-ideaOrange focus:outline-none focus:text-white focus:bg-ideaOrange transition duration-150 ease-in-out"
    href="/no-active"
  >
    Profile
  </a>
</DocumentFragment>
`;

exports[`MobileLink should be rendered correctly when current is enabled 1`] = `
<DocumentFragment>
  <span
    class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-white bg-ideaOrange focus:outline-none focus:text-white focus:bg-ideaOrange transition duration-150 ease-in-out"
  >
    Profile
  </span>
</DocumentFragment>
`;
